### PR TITLE
feat: update withdraw

### PR DESCRIPTION
**Pull request by royvardhan**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new `emergencyWithdraw` function for handling token withdrawals more efficiently.
- **Improvements**
	- Updated token withdrawal logic to include robust error handling for scenarios like zero addresses and insufficient balances.
- **Refactor**
	- Changed event names within the `TokenFactory` contract for better clarity and consistency.

### CODE REVIEW

File: contracts/TokenFactory.sol [hunk 1]
```diff
@@ -397,17 +397,22 @@ contract TokenFactory is Ownable {
...
-        IERC20 _usdc = IERC20(USDC);
-        _usdc.transfer(_to, _usdc.balanceOf(address(this)));
-        emit AccumulatedFeesWithdrawn(_to, _usdc.balanceOf(address(this)));
+        IERC20 token = IERC20(_token);
+        uint256 balance = token.balanceOf(address(this));
+        if (balance == 0) {
+            revert TokenFactory__InsufficientBalance();
+        }
+        token.transfer(_to, balance);
+        emit AccumulatedTokenWithdrawn(_to, token.balanceOf(address(this)));
```
Comment (mejiasd3v) [inline contracts/TokenFactory.sol]: Why is this called Accumulated?

File: contracts/TokenFactory.sol [hunk 2]
```diff
@@ -397,17 +397,22 @@ contract TokenFactory is Ownable {
...
-     * @dev Withdraws any remaining USDC fees to the specified address.
-     * @param _to Address to which the remaining fees are withdrawn.
+     * @dev Withdraws any stuck tokens (LP Or USDC) to the specified address.
+     * @param _token Address of the token to be withdrawn.
+     * @param _to Address to which the tokens are withdrawn.
      */
 
-    function withdrawFee(address _to) external onlyOwner {
-        if (_to == address(0)) {
+    function withdraw(address _token, address _to) external onlyOwner {
```
Comment (mejiasd3v) [inline contracts/TokenFactory.sol]: I've seen this generally called emergencyWithdraw I would rename it

**Review by mejiasd3v — APPROVED** [review]
Minor comments, otherwise looks good

File: contracts/TokenFactory.sol (hunk 2 above)
Comment (royvardhan) [reply contracts/TokenFactory.sol]: Done

**Review by mejiasd3v — APPROVED** [review]

**Pull request merged**